ocean_lib/web3_internal/wallet.py

```diff
@@ -123,20 +123,35 @@ def sign_tx(
             )
         else:
             nonce = Wallet._get_nonce(self.web3, account.address)
+            from ocean_lib.web3_internal.contract_base import ContractBase
+
+            if not gas_price:
+                try:
+                    history = self.web3.eth.fee_history(
+                        block_count=1, newest_block="latest"
+                    )
+                    tx["type"] = "0x2"
+                    fee_tx = ContractBase.get_max_fee_per_gas(
+                        web3=self.web3, tx=tx, history=history
+                    )
+                    tx.update(fee_tx)
+                    tx["gas"] = GAS_LIMIT_DEFAULT
+                except ValueError as e:
```

Comment:
the duplication here will be fixed when you move history inside the get_max_fee_per_gas function.